src/ui/FieldValue/IFieldValueCompatibleFacet.ts

```diff
@@ -1,10 +1,11 @@
 import { Component } from '../Base/Component';
 
 export function isFacetFieldValueCompatible(facet: Component): facet is IFieldValueCompatibleFacet {
-  return !!(facet['hasSelectedValue'] && facet['selectValue'] && facet['deselectValue']);
+  return !!facet['isFieldValueCompatible'];
 }
 
 export interface IFieldValueCompatibleFacet extends Component {
```

Comment:
I think we should ensure that all facets are implementing this interface. It think it's very easy for someone working on a new facet component to forget to include this property.

````
export class FacetRange extends Facet implements IComponentBindings, IFieldValueCompatibleFacet   {
````

Also, now that the interface has an `isFieldValueCompatible` property, are the other properties needed? or could they be removed?

Reply:
Yes, it's already the case, DynamicFacet & Facet both implement this interface since they are both compatible with FieldValue.
FacetRange & DynamicFacetRange extends those facets, so they inherit the properties, it was necessary to add a flag so they don't.
> Also, now that the interface has an isFieldValueCompatible property, are the other properties needed? or could they be removed?
isFieldValueCompatible is not enough, we must ensure that possess the methods that FieldValue uses, it's safer this way.

Reply:
If you create a new Facet component, and want to make is compatible with FieldValue, you have to extend IFieldValueCompatibleFacet, and it will force you to implement the necessary methods, that's the idea behind it.